Sources/NIOFileSystem/DirectoryEntry.swift

```diff
@@ -21,6 +21,10 @@ public struct DirectoryEntry: Sendable, Hashable, Equatable {
     /// - Precondition: The path must have at least one component.
     public let path: FilePath
 
+    public var filePath: NIOFilePath {
```

Comment:
Docs please! Also, I think we should store the `NIOFilePath` and have a computed property for `path`.